src/helpers/Move.js

```diff
@@ -28,21 +29,130 @@ export const move = (move, position) => {
+            case 'e':
+                columnInd = 4;
+                break;
+            case 'f':
+                columnInd = 5;
+                break;
+            case 'g':
+                columnInd = 6;
+                break;
+            case 'h':
+                columnInd = 7;
+                break;
+            default:
+                columnInd = 0;
+                break;   
+        }
+
+        if(players[currentPlayer] === 'white'){
+            
+            if(piece === 'Q') {              
+                let allQueens = fields.filter(field => field.piece === 'Q');
+                allQueens[0].piece = '';
+                let newColumnFields = fields.filter(field => field.column === moveColumn);
+                let newPosition = newColumnFields.filter(field => field.row === Number(moveRow));
+                newPosition[0].piece = 'Q';
+            } else if(piece === 'K') {
+                let allKings = fields.filter(field => field.piece === 'K');
+                allKings[0].piece = '';
+                let newColumnFields = fields.filter(field => field.column === moveColumn);
+                let newPosition = newColumnFields.filter(field => field.row === Number(moveRow));
+                newPosition[0].piece = 'K';
+            } else if(piece === 'R'){
+                let allRooks = fields.filter(field => field.piece === 'R');
+                let goodRook = allRooks.filter(rook => (rook.column === moveColumn || rook.row === moveRow));
+                goodRook[0].piece = '';
+                let newColumnFields = fields.filter(field => field.column === moveColumn);
+                let newPosition = newColumnFields.filter(field => field.row === Number(moveRow));
+                newPosition[0].piece = 'R';
+            } else if(piece === 'N'){ 
+                let eightPositions = fields.filter(field => ((field.columnIndex === (columnInd - 2)) && (field.row === (Number(moveRow) + 1))) ||
+                                                            ((field.columnIndex === (columnInd - 2)) && (field.row === (Number(moveRow) - 1))) ||
+                                                            ((field.columnIndex === (columnInd - 1)) && (field.row === (Number(moveRow) + 2))) ||
+                                                            ((field.columnIndex === (columnInd - 1)) && (field.row === (Number(moveRow) - 2))) ||
+                                                            ((field.columnIndex === (columnInd + 1)) && (field.row === (Number(moveRow) + 2))) ||
+                                                            ((field.columnIndex === (columnInd + 1)) && (field.row === (Number(moveRow) - 2))) ||
+                                                            ((field.columnIndex === (columnInd + 2)) && (field.row === (Number(moveRow) + 1))) ||
+                                                            ((field.columnIndex === (columnInd + 2)) && (field.row === (Number(moveRow) - 1))) );
```

Comment:
I think that you schould call logic from outside helper to see if new field is valid for particular movement.